code/modules/vending/_vending.dm

```diff
@@ -369,6 +377,29 @@ GLOBAL_LIST_EMPTY(vending_products)
 	if(panel_open && is_wire_tool(I))
 		wires.interact(user)
 		return
+
+	if(custom)
```

Comment:
unless im reading this wrong you should add a last return so you don't check for the other refill canisters as you can just return always at the end of this logic (if there was a private account linked, but the user wasn't the owner)